db/db_test_util.cc

```diff
@@ -496,6 +496,13 @@ Status DBTestBase::ReadOnlyReopen(const Options& options) {
 
 Status DBTestBase::TryReopen(const Options& options) {
   Close();
+  last_options_.table_factory.reset();
```

Comment:
How about (in a separate PR) make the last_options_ a unique_ptr, and destruct the content before assignment?